onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -723,14 +851,22 @@
 }
 
 Status TransposeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
-  auto& shaper(model_builder.GetShaper());
   const auto& initializers(model_builder.GetInitializerTensors());
-
+  const auto& graph_viewer = model_builder.GetGraphViewer();
   const auto& input = node_unit.Inputs()[0].node_arg.Name();
   const auto& output = node_unit.Outputs()[0].node_arg.Name();
+
   NodeAttrHelper helper(node_unit);
+  auto& shaper(model_builder.GetShaper());
+
+  const auto* input_node_arg = graph_viewer.GetNodeArg(input);
+  ORT_RETURN_IF(input_node_arg == nullptr, "Invalid node_arg for input: ", input);
+
+  const auto* shape_proto = input_node_arg->Shape();
+  ORT_RETURN_IF(shape_proto == nullptr, "shape_proto cannot be null for: ", input_node_arg->Name());
+
```

Comment:
Wondering if these checks are worth the binary size cost.

If we're always looking up the NodeArg via an input/output name for a Node that should _always_ exist. Otherwise the Graph infrastructure that allowed the Node instance to be created is fundamentally broken.

Previously it looks like we didn't check that Shape() was nullptr here as we just got a value from shaper with no error path. Was that incorrect, or would we never get to this point if the shape was nullptr? Naively I would have expected the op support checker to require a shape given the lack of support for dynamic shapes in NNAPI.

Reply:
Right. Just checked actually in Baseopsupportchecker::HasSupportedInputsOutputs, it already checks shape_proto is not null, also in each individual Opsupportchecker, it calls GetShape() helper, also requires shape to present.

so will never get to this point if the shape is a nullptr. removed the redundant check here.